install/kubernetes/helm/istio/values.yaml

```diff
@@ -191,6 +191,9 @@ global:
   # distributed through the SecretDiscoveryService instead of using K8S secrets to mount the certificates.
   sdsEnabled: false
 
+  # indicate the domain used in SPIFFE identity URL
+  identityDomain: cluster.local 
```

Comment:
@gyliu513 I think that the clusters dns suffix is not really related to the spiffe id that Citadel should use to express cluster identity.

Reply:
Thanks @loewenstein , but I think they are same, the clusters dns suffix was also used by Citadel. @elevran comments?

Reply:
@gyliu513 they happened to be the same value, but represent different things. I think citadel just takes "cluster.local" for convenience.

Correct me if I'm wrong @myidpt

Reply:
Thanks for bringing this up. +1 to incfly and loewenstein.

First, they mean two different things.
DNS prefix is used in K8s&Istio service naming. Service names are used in DNS lookup and Istio service discovery.
Identity domain is only used in Istio identities (Citadel picked the value "cluster.local" initially to align with the K8s naming scheme, but it could be any string), they are used in service to service authentication.
So technically DNS prefix and identity domain can be different.

Second, does it make sense to align DNS prefix and identity domain? I discussed this with @costinm , and I think Costin is in favour of aligning them. I'm OK with that, but are there legitimate use cases we need them to be different?

Reply:
@costinm what's the reasoning for aligning dns and spiffe domains? I guess dns could be a good default spiffe trust domain for single cluster. However, as soon as you put a second cluster into the picture one might want to identify traffic origin, but there is still no need to have separate local dns domains. Further, the Istio operator might not even be in control of the cluster dns.

Reply:
Thanks @loewenstein , I get your point, so you mean if there are multiple clusters, then those clusters can still share one dns domain but use different security domain? This seems to be a valid case to support your PR, as the Istio operator might not even be in control of the cluster dns. 

Let us wait some comments from @costinm

Reply:
+1 to loewenstein